src/libmeasurement_kit/common/utils_impl.hpp

```diff
@@ -0,0 +1,63 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef SRC_LIBMEASUREMENT_KIT_COMMON_UTILS_IMPL_HPP
+#define SRC_LIBMEASUREMENT_KIT_COMMON_UTILS_IMPL_HPP
+
+#include "../common/utils.hpp"
+#include <cstdio>
+#include <measurement_kit/common.hpp>
+
+namespace mk {
+
+/*
+ * There are many ways in C++ to slurp the whole content of a file into a
+ * C++ string. The following is my favourite. Compared to other methods,
+ * not only this seems fastest, but also it provides greater control over
+ * errors that can occur. Thus mocking is simpler.
+ *
+ * See <http://stackoverflow.com/questions/116038/what-is-the-best-way-to-read-an-entire-file-into-a-stdstring-in-c>.
+ */
+template <typename T, MK_MOCK_NAMESPACE(std, fopen),
+          MK_MOCK_NAMESPACE_SUFFIX(std, fseek, 1),
+          MK_MOCK_NAMESPACE(std, ftell),
+          MK_MOCK_NAMESPACE_SUFFIX(std, fseek, 2),
+          MK_MOCK_NAMESPACE(std, fread), MK_MOCK_NAMESPACE(std, fclose)>
+ErrorOr<std::vector<T>> slurpv_impl(std::string p) {
+    FILE *filep = std_fopen(p.c_str(), "rb");
+    if (filep == nullptr) {
+        return FileIoError();
+    }
+    if (std_fseek_1(filep, 0, SEEK_END) != 0) {
+        std_fclose(filep);
+        return FileIoError();
+    }
+    // Note: ftello() might be better for reading very large files but
```

Comment:
[maybe bikeshedding] Considering the fact that we are working with small files, should we check somehow that the files are really small. Is my understanding right that if you try to read a file of 50mb with this function you will get a vector of 50mb in memory? If yes, should we limit this size of the file (checking the returned fseek position), considering that we are working on mobile devices??